packages/perpswap/src/contracts/vault.rs

```diff
@@ -0,0 +1,130 @@
+#![allow(missing_docs)]
+use cosmwasm_std::{Addr, Uint128};
+
+// Message to instantiate the contract
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+pub struct InstantiateMsg {
+    // Denomination of the USDC token
+    pub usdc_denom: String,
+
+    // Address of the factory (as string, validated later)
+    pub factory_address: String,
+
+    // Address of the USDCLP contract (as string, validated later)
+    pub usdclp_address: String,
+
+    // Governance address (as string, validated later)
+    pub governance: String,
+
+    // Initial list of operators (as strings, validated later)
+    pub initial_operators: Vec<String>,
+
+    // Initial allocation percentages to markets
+    pub yield_allocation_bps: Vec<u16>,
+}
+
+// Configuration structure for the vault
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct Config {
+    // Denomination of the USDC token (e.g., "uusdc")
+    pub usdc_denom: String,
+
+    // Address of the factory contract that manages markets
+    pub factory_address: Addr,
+
+    // Address of the CW20 contract handling USDCLP tokens
+    pub usdclp_address: Addr,
+
+    // Address authorized for critical actions (like pausing the contract)
+    pub governance: Addr,
+
+    // List of addresses authorized for specific operations
+    pub operators: Vec<Addr>,
+
+    // Allocation percentages to markets in basis points (100 bps = 1%)
+    pub yield_allocation_bps: Vec<u16>,
+}
+
+// Executable messages for the contract
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    // Deposit USDC and receive USDCLP
+    Deposit {
+        amount: Uint128,
```

Comment:
The amount shouldn't be present here, instead we should determine the amount by looking at the submitted funds.